src/library_glfw.js

```diff
@@ -88,6 +90,9 @@ var LibraryGLFW = {
     '$FS',
 #endif
   ],
+  $GLFW__postset: `
+    // making setHiDPIAware public for users of glfw (dynamic switching)
+    Module["setHiDPIAware"] = GLFW.setHiDPIAware;`,
```

Comment:
I think you can just do this:

```
setHiDPIAware: GLFW.setHiDPIAware,
```

Then users who want to export that function can do so like any other library function.   No need to export it by default like this.

Reply:
I don't understand what you are suggesting:

1) if I write `setHiDPIAware: GLFW.setHiDPIAware` isn't it defining a field referring to the same field?
2) how does a user gets access to this function? can you explain what a user would have to do to access it? Right now I write this (in C) `EM_ASM({Module.setHiDPIAware($0);}, isHiDPIAware ? 1 : 0);`. What would I write after the change and is there compilation flags to be provided?

Reply:
`$setHiDPIAware: GLFW.setHiDPIAware,` would be something you would write at the higher level (sorry I forgot to include the leading dollar sign the first time) not within the `GLFW` object.  It would act like any other JS library function.

You can include JS library functions using `-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE` on the command line and export them via `-sEXPORTED_RUNTIME_FUNCTIONS` or `-sEXPORTED_FUNCTIONS`.

If you want to call `setHiDPIAware` from an EM_ASM block you wouldn't actually need to the `Module.` prefix at all.   In that case you can just call `setHiDPIAware` and then specify it as a dependency using `EM_JS_DEPS(foo, "$setHiDPIAware");` in your C/C+ source code.

Reply:
> You can include JS library functions using -sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE on the command line and export them via -sEXPORTED_RUNTIME_FUNCTIONS or -sEXPORTED_FUNCTIONS.

Are these options on the client side or something I need to change on the emscripten side?

Is it wise to call it `$setHiDPIAware`? Should it be called `$glfwSetHiDPIAware` or `$glfwEmscriptenSetHiDPIAware` so that there are less risks of collision?

Reply:
> > > You can include JS library functions using -sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE on the command line and export them via -sEXPORTED_RUNTIME_FUNCTIONS or -sEXPORTED_FUNCTIONS.
> > 
> > 
> > Are these options on the client side or something I need to change on the emscripten side?
> 
> Those options are what end uses would use if they want to call this function.

(Just like any other JS function)

Reply:
If you call a JS function from C then you don't need to do anything, it will automatically get included.

If you want to call a JS function on the `Module` object, or from and `EM_JS` block then you need to tell emscripten to include it somehow, otherwise it won't be included in the generated code.

Are you expecting end users to need to call `glfwSetHiDPIAware`, and if so from where?   For JS functions exposed to C/C++ they wouldn't need to do anything.  For JS functions exposed to the outside on the `Module.` object they would need to specify this function in a command line flag, just like any other JS library function.

Reply:
I believe the ability to change dynamically either from C or Javascript is desirable:

* from C/C++, I envision it from a menu/checkbox (for example with ImGui I could provide both)
* from Javascript, the "shell" allow to request fullscreen for example, and so changing HiDPI in a similar manner could be an option.

If I have to keep one, I would pick C/C++

I am going to investigate your suggestions and see where that leads me.

Reply:
In general you don't need to pick one.. a JS library function can be called from both C/C++ and from external JS code.

The important thing is that if you want to call a function from outside the module you need to export it via a command line flag.   This is true for both native and JS function actually.  I general, if don't ask (via the command line) for a function to be exported it won't be.

Reply:
For now I have pushed a change that only rename the function exported in Module to make it more specific to glfw.

```
  $GLFW__postset: `
    // making glfwSetHiDPIAware public for users of glfw (dynamic switching)
    Module["glfwSetHiDPIAware"] = GLFW.setHiDPIAware;`,
```

Reply:
I am trying the export/import method and can't seem to make it work:

in `library_gflw.js` I added:

```javascript
  glfwSetHiDPIAware: (isHiDPIAware) => GLFW.setHiDPIAware(isHiDPIAware),
```

Now on the client side (C code), I added this:

```cpp
extern void glfwSetHiDPIAware(bool);
// ....
        glfwSetHiDPIAware(isHiDPIAware);
```

and the compilation flags:

```
-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE=glfwSetHiDPIAware -sEXPORTED_FUNCTIONS=glfwSetHiDPIAware
```

and getting this error:

```text
wasm-ld: error: CMakeFiles/imgui_raylib_cmake.dir/main.cpp.o: undefined symbol: glfwSetHiDPIAware(bool)
```

I guess I must be missing...

Reply:
You need to wrap the declaration in `extern "C"` which all JS library functions are C symbols, not C++.

Also you shouldn't need any extra command line flag to call it from C/C++.

Reply:
See my most recent update about the changes